tests/slots.rs

```diff
@@ -163,3 +163,20 @@ fn slots_layer2() {
   drop(es_isolate);
   assert_eq!(drop_count.load(Ordering::SeqCst), 2);
 }
+
+#[test]
+fn remove_slot() {
+  let drop_count = Rc::new(AtomicUsize::new(0));
+  let mut core_isolate = CoreIsolate::new(drop_count.clone());
+
+  let state: CoreIsolateState =
+    core_isolate.remove_slot::<CoreIsolateState>().unwrap();
+  drop(core_isolate);
+
+  // The state should not be dropped with the isolate because we own it now.
+  assert_eq!(drop_count.load(Ordering::SeqCst), 0);
+
+  // We're dropping it now on purpose.
+  drop(state);
+  assert_eq!(drop_count.load(Ordering::SeqCst), 1);
+}
```

Comment:
Can you rework the test a bit to first add struct to a slot, increment it, remove the slot and verify that following calls to get the slot fail?

Reply:
Sure, give me a few minutes